code/go/0chain.net/miner/protocol_block.go

```diff
@@ -591,7 +591,31 @@ func (mc *Chain) updateFinalizedBlock(ctx context.Context, b *block.Block) {
 	for _, txn := range b.Txns {
 		txns = append(txns, txn)
 	}
+
+	tii := newTxnIterInfo(mc.BlockSize())
+	invalidTxns := tii.checkForInvalidTxns(b.Txns)
+
 	transaction.RemoveFromPool(ctx, txns)
+	
+	if len(invalidTxns) > 0 {
+		transaction.RemoveFromPool(ctx, invalidTxns)
+	}
+}
+
+func (tii *TxnIterInfo) checkForInvalidTxns(txns []*transaction.Transaction) []datastore.Entity {
+	invalidTxns := []datastore.Entity{}
+	pastTxns := tii.pastTxns
```

Comment:
This `tii.pasTxns` is empty if this `tii` is created in `updateFinalizedBlock`, therefore the invalidTxns below will always be empty. The issue #1249 is actually fixed, we should have closed it to avoid ambiguous.